From 81a80c9973833f7cd93dc83ce0f23572dd81befe Mon Sep 17 00:00:00 2001
From: Min Liu <minliu@codeaurora.org>
Date: Wed, 28 Mar 2018 13:20:13 +0800
Subject: qcacld-2.0: Fix UAF in WLAN HDD

UAF issue is in function hdd_open_adapter:

pAdapter is freed by calling free_netdev(pAdapter->dev),
but pAdapter->macAddressCurrent.bytes is referenced afterwards.

Fix the issue by swapping these two statements.

Change-Id: I6910a56f9a204fdd8eaad54d8443867ee6a37fdb
CRs-Fixed: 2213280
---
 CORE/HDD/src/wlan_hdd_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CORE/HDD/src/wlan_hdd_main.c b/CORE/HDD/src/wlan_hdd_main.c
index 773751b..e05d2e3 100644
--- a/CORE/HDD/src/wlan_hdd_main.c
+++ b/CORE/HDD/src/wlan_hdd_main.c
@@ -11730,9 +11730,9 @@ err_init_adapter_mode:
 err_init_packet_filtering:
 	hdd_adapter_runtime_suspend_denit(adapter);
 
-	free_netdev(adapter->dev);
 	wlan_hdd_release_intf_addr(hdd_ctx,
 				   adapter->macAddressCurrent.bytes);
+	free_netdev(adapter->dev);
 
 	/* If bmps disabled enable it */
 	if (!hdd_ctx->cfg_ini->enablePowersaveOffload) {
-- 
cgit v1.1

